### PR TITLE
REST/Websocket API deep refactoring

### PR DESCRIPTION
**Pull request by siy**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/org/radix/Radix.java [hunk 1]
```diff
@@ -17,33 +17,32 @@
...
+import com.radixdlt.utils.MemoryLeakDetector;
 
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URISyntaxException;
 import java.security.Security;
+import java.util.Map;
 import java.util.Properties;
 
 public final class Radix {
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/org/radix/Radix.java:48]: Doesn't seem like there are any functional changes here. Would be better to not include cosmetic-only changes.

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RestUtils.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import java.util.function.Supplier;
+
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.Headers;
+import io.undertow.util.StatusCodes;
+
+public class RestUtils {
+	public static final String CONTENT_TYPE_JSON = "application/json";
+
+	private RestUtils() {}
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RestUtils.java:41]: Would be good to throw `IllegalStateException` on instantiation.

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RestUtils.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import java.util.Deque;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.Headers;
+import io.undertow.util.StatusCodes;
+
+public class RestUtils {
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RestUtils.java:38]: Add `final` so can't be extended.

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RpcController.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.google.common.annotations.VisibleForTesting;
+
+import io.undertow.Handlers;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.server.RoutingHandler;
+
+import static org.radix.api.http.RestUtils.respondAsync;
+
+public class RpcController {
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RpcController.java:30]: Add `final`

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/http/SystemController.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+import io.undertow.server.HttpServerExchange;
+import io.undertow.server.RoutingHandler;
+
+import static org.radix.api.http.RestUtils.respond;
+import static org.radix.api.http.RestUtils.withBodyAsync;
+import static org.radix.api.jsonrpc.JsonRpcUtil.jsonArray;
+import static org.radix.api.jsonrpc.JsonRpcUtil.jsonObject;
+
+public class SystemController {
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/org/radix/api/http/SystemController.java:39]: Add `final`

File: radixdlt-core/radixdlt/src/main/java/org/radix/network/discovery/BootstrapDiscovery.java [hunk 6]
```diff
@@ -17,55 +17,52 @@
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/org/radix/network/discovery/BootstrapDiscovery.java:17]: This seems unrelated to API. Would be better to have this in a different PR.

File: radixdlt-core/radixdlt/src/main/java/org/radix/network/discovery/Whitelist.java [hunk 7]
```diff
@@ -29,18 +29,16 @@
 import org.apache.logging.log4j.Logger;
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/org/radix/network/discovery/Whitelist.java:29]: Unrelated PR with cosmetic only changes.

**Review by talekhinezh — APPROVED** [review]

File: radixdlt-core/radixdlt/src/main/java/org/radix/Radix.java (hunk 1 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/org/radix/Radix.java:48]: Rolled back changes. For these minor changes I'll create a separate PR.

File: radixdlt-core/radixdlt/src/main/java/org/radix/network/discovery/BootstrapDiscovery.java (hunk 6 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/org/radix/network/discovery/BootstrapDiscovery.java:17]: Rolled back changes. For these minor changes I'll create a separate PR.

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/http/SystemController.java (hunk 5 above)
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/org/radix/api/http/SystemController.java:39]: Done

**Review by LukasGasior1 — APPROVED** [review]
LGTM

File: radixdlt-core/radixdlt/src/main/java/org/radix/Radix.java [hunk 8]
```diff
@@ -129,43 +128,45 @@ public static void main(String[] args) {
...
 
-		final SystemInfoRunner infoStateRunner = injector.getInstance(SystemInfoRunner.class);
+		var infoStateRunner = injector.getInstance(SystemInfoRunner.class);
 		infoStateRunner.start();
 
-		final Map<String, ModuleRunner> moduleRunners = injector.getInstance(Key.get(new TypeLiteral<Map<String, ModuleRunner>>() { }));
+		var moduleRunners = injector.getInstance(Key.get(new TypeLiteral<Map<String, ModuleRunner>>() {}));
 
-		final ModuleRunner syncRunner = moduleRunners.get("sync");
+		var syncRunner = moduleRunners.get("sync");
```
[the remark this answers is not in the document]
Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/org/radix/Radix.java]: after some considerations I've decided that `final` for local variables is unnecessarily verbose.
